include/nbl/video/IGPUCommandPool.h

```diff
@@ -31,11 +35,20 @@ class NBL_API IGPUCommandPool : public core::IReferenceCounted, public IBackendO
         // Vulkan: const VkCommandPool*
         virtual const void* getNativeHandle() const = 0;
 
+        virtual void reset()
+        {
+            // Increase reset counter
+            m_resetCount.fetch_add(1);
+        }
+
     protected:
         virtual ~IGPUCommandPool() = default;
 
         core::bitflag<E_CREATE_FLAGS> m_flags;
         uint32_t m_familyIx;
+        std::atomic_uint32_t m_resetCount;
+
+        std::atomic_uint32_t& getResetCounter() { return m_resetCount; }
```

Comment:
you can make it public, so you don't need the friendship and forward declaration of `IGPUCommandBuffer`

Also its better to not return a reference and just "do the load"
```cpp
inline uint32_t getResetCount() const {return m_resetCount.load();}
```

Reply:
Resolved.